src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -228,6 +267,186 @@ TimeSynchronizationServer & TimeSynchronizationServer::Instance()
     return sTimeSyncInstance;
 }
 
+TimeSynchronizationServer::TimeSynchronizationServer() :
+#if TIME_SYNC_ENABLE_TSC_FEATURE
+    mOnDeviceConnectedCallback(OnDeviceConnectedWrapper, this),
+    mOnDeviceConnectionFailureCallback(OnDeviceConnectionFailureWrapper, this),
+#endif
+    mOnTimeSyncCompletion(OnTimeSyncCompletionWrapper, this), mOnFallbackNTPCompletion(OnFallbackNTPCompletionWrapper, this)
+{}
+
+void TimeSynchronizationServer::AttemptToGetFallbackNTPTimeFromDelegate()
+{
+    // Sent as a char-string to the delegate so they can read it easily
+    char defaultNTP[kMaxDefaultNTPSize];
+    MutableCharSpan span(defaultNTP);
+    if (GetDefaultNtp(span) != CHIP_NO_ERROR)
+    {
+        emitTimeFailureEvent(kRootEndpointId);
+        return;
+    }
+    if (span.size() > kMaxDefaultNTPSize)
```

Comment:
How could this happen, short of the callee writing out of bounds (in which case we are already in trouble)?

Reply:
It shouldn't but I figured it wouldn't hurt to check. Checking span sizes generally just seems like good practice.

Reply:
I guess it makes it look like if the size check fails you are "ok" whereas in fact you've already been owned... but ok.